test/runnable/mars1.d

```diff
@@ -1675,6 +1675,23 @@ void testdivcmp()
     assert(foo(5, 4));
 }
 
+////////////////////////////////////////////////////////////////////////
+
+// https://issues.dlang.org/show_bug.cgi?id=16189
+
+void test16189()
+{
+    ubyte[9][1] data;
+    uint a = 0;
+  loop:
+    data[0] = data[a];
+    a--;
+    bool b = false;
+    if (b) goto loop;
+    assert(a == -1); // Fails with -O
```

Comment:
Should this comment still be here? I assume the assert does *not* fail now.

Reply:
Past tense "Failed"?  Doesn't bother me really.